### PR TITLE
[Tests] Add combined pipeline tests

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

The PR adds tests for the loading logic of connected pipelines and thereby fixes: #4289

### CODE REVIEW

File: tests/pipelines/kandinsky_v22/test_kandinsky.py [hunk 1]
```diff
@@ -183,6 +183,9 @@ def test_kandinsky(self):
 
         pipe = self.pipeline_class(**components)
         pipe = pipe.to(device)
+        import ipdb
+
+        ipdb.set_trace()
```
Comment (patrickvonplaten) [inline tests/pipelines/kandinsky_v22/test_kandinsky.py:188]:
```suggestion
```

File: tests/pipelines/test_pipelines_combined.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+from diffusers import DiffusionPipeline, KandinskyV22CombinedPipeline, KandinskyV22Pipeline, KandinskyV22PriorPipeline
+from diffusers.pipelines.pipeline_utils import CONNECTED_PIPES_KEYS
+
+
+def state_dicts_almost_equal(sd1, sd2):
+    sd1 = dict(sorted(sd1.items()))
+    sd2 = dict(sorted(sd2.items()))
+
+    models_are_equal = True
+    for ten1, ten2 in zip(sd1.values(), sd2.values()):
```
Comment (sayakpaul) [inline tests/pipelines/test_pipelines_combined.py:30]: Nothing actionable, but what is `ten`?

File: tests/pipelines/test_pipelines_combined.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        if (ten1 - ten2).abs().sum() > 1e-3:
+            models_are_equal = False
+
+    return models_are_equal
+
+
+class CombinedPipelineFastTest(unittest.TestCase):
+    def modelcard_has_connected_pipeline(self, model_id):
```
Comment (sayakpaul) [inline tests/pipelines/test_pipelines_combined.py:38]: Damn! I would have bever thought of testing it like that.

File: tests/pipelines/test_pipelines_combined.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import torch
+from huggingface_hub import ModelCard
+
+from diffusers import DiffusionPipeline, KandinskyV22CombinedPipeline, KandinskyV22Pipeline, KandinskyV22PriorPipeline
+from diffusers.pipelines.pipeline_utils import CONNECTED_PIPES_KEYS
```
Comment (sayakpaul) [inline tests/pipelines/test_pipelines_combined.py:22]: Should the keys be put into `constants`?

File: tests/pipelines/test_pipelines_combined.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                comp = prior_comps[k]
+            else:
+                comp = decoder_comps[k]
+
+            if isinstance(component, torch.nn.Module):
+                assert state_dicts_almost_equal(component.state_dict(), comp.state_dict())
+            elif hasattr(component, "config"):
+                assert dict(component.config) == dict(comp.config)
+            else:
+                assert component.__class__ == comp.__class__
```
Comment (sayakpaul) [inline tests/pipelines/test_pipelines_combined.py:103]: Clean!

**Review by sayakpaul — APPROVED** [review]
Very nice tests! Thanks for adding.

File: tests/pipelines/test_pipelines_combined.py (hunk 2 above)
Comment (yiyixuxu) [reply tests/pipelines/test_pipelines_combined.py:30]: tensor? just a wild guess 😁

**Review by yiyixuxu — APPROVED** [review]
super cool!

**Pull request merged**